lib/hashes.md

```diff
@@ -5,40 +5,46 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A hash is a group of items in which each item has an attribute or value stored along with it.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+Anything that has a value beyond its own name (as far as I know now, these are usually numerical values)--examples include price lists, quantities of attendees for a group of events, run times of various programs, etc.
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+Hashes and arrays are similar in that they're both a group of items that will be either listed or sorted together. While arrays are simply a list of names, items, or numbers, hashes include more information, as each item can have an additional value attached to it.
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+hashname ["itemname"] will call the value of that item.
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+hashname ["itemname"] = itemvalue
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+This calls for an .each do operation--this is the example that rubymonk.com gives to raise the menu prices by 10%:
+restaurant_menu = { "Ramen" => 3, "Dal Makhani" => 4, "Coffee" => 2 }
+restaurant_menu.each do |item, price|
+  restaurant_menu[item] = price + (price * 0.1)
+end
+
+I didn't get this code right when I was attempting it for myself, but after seeing their cheat solution, it makes sense.
 
```

Comment:
The original hash contains a restaurant menu in which the three items cost $3, $4, and $2. The 'item' variable refers to the name of each food on the menu, while the 'price' variable is the cost of each of these foods. The each.do operation is multiplying each of these prices by 1.1 in order to raise them by 10%. 

I am still a little confused about how the variables within the || brackets are established. I'm guessing that because there are two variables here, the first automatically refers to the key, and the second automatically refers to the value? My google searches have given more examples, but this still isn't totally clear in my mind.